src/TypeScript/EslintBridgeClient/EslintBridgeProcess.cs

```diff
@@ -162,17 +180,23 @@ private void Stop()
         {
             logger.LogDebug(Resources.INFO_TerminatingServer);
 
+            if (Process != null)
+            {
+                Process.ErrorDataReceived -= OnErrorDataReceived;
+                Process.OutputDataReceived -= OnOutputDataReceived;
+                Process.Exited -= Process_Exited;
+            }
             if (Process == null || Process.HasExited)
             {
                 logger.LogDebug(Resources.INFO_ServerAlreadyTerminated);
+            }
+            else
```

Comment:
Calling `Process?.Dispose();` on a terminated process can actually throw, so we shouldn't have been calling it all the time

Reply:
There's still a race condition here, isn't there? The process could have finished after the `HasExited` check but before the code in the `else` is executed ([the CFamily process-handling code](https://github.com/SonarSource/sonarlint-visualstudio/blob/6b9a8a7623906def168229d9089bf6e17a9c00a4/src/Integration.Vsix/CFamily/ProcessRunner.cs#L167) guards against this).

I don't think there is a safe way of cleaning up a process that doesn't involve exception handling.